### PR TITLE
feat: dcmaw-8571 privacy policy link button

### PR DESCRIPTION
**Pull request by jdubey**
# DCMAW-8571: iOS | Add privacy policy link to analytics opt-in page

Add the privacy policy link to the Analytics opt-in page.  This leverages changes made in mobile-ios-common

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [x] Met all accessibility requirements?
    - [x] Checked dynamic type sizes are applied
    - [x] Checked VoiceOver can navigate your new code
    - [x] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [x] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [x] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

File: OneLogin.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2055,11 +2060,11 @@
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_TEAM = N8W395F695;
+				EXTERNAL_BASE_URL = signin.account.gov.uk;
 				FEATURE_FLAG_FILE = FeatureFlagsBuild;
 				GENERATE_INFOPLIST_FILE = YES;
 				INFOPLIST_FILE = "OneLogin-Info.plist";
-				INFOPLIST_KEY_CFBundleDisplayName = "One Login - Build";
-				INFOPLIST_KEY_NSCameraUsageDescription = "This app does not use the camera";
```
Comment (DylanRJ) [inline OneLogin.xcodeproj/project.pbxproj:2062]: Looks like the Xcode project file is doing some annoying stuff here in terms of renaming the app display name and getting rid of the camera usage description. I've seen this happen locally as well, very annoying.
Worth keeping an eye out for changes in this project file, we've fallen foul of stuff like this on the program in the past with stuff making it's way into production... less of a concern for this app atm but good practice to get into 🙂
With bringing these project settings back, there's a nice way of toggling settings names/titles in Xcode (if you don't know about it already).
When you're in the project file, Editor -> Show Setting Names will show them in the format you see here and may make it a little easier
image 2024-04-22 at 14 31 56

File: OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (DylanRJ) [inline OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: I think we should be careful about updating these transitive dependencies where they haven't come from updating our top-level packages. I've not heard a definitive answer on this but ideally we need our packages' `Package.resolved` files to match this file

Comment (jdubey) [reply OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: So if I understand what you are saying, every package that is dependent on mobile-ios-common should get updated to this new revision?

File: OneLogin.xcodeproj/project.pbxproj (hunk 1 above)
Comment (mcgarj) [reply OneLogin.xcodeproj/project.pbxproj:2062]: Is the only way to change these to revert them manually to what they are in the commit changes? And there's no way to stop Xcode randomly doing this?

**Review by DylanRJ — APPROVED** [review]

**Review by mcgarj — APPROVED** [review]

File: OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (DylanRJ) [reply OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: Hopefully dependabot will resolve these versions to what the top-level package dependencies have in their `Package.resolved` files.
If not, we probably need to think about how we can manage this going forward

**Pull request merged**